ArduSub/failsafe.cpp

```diff
@@ -499,3 +499,55 @@ void Sub::failsafe_terrain_act()
         arming.disarm(AP_Arming::Method::TERRAINFAILSAFE);
     }
 }
+
+#if AP_SUB_RC_ENABLED
+void Sub::set_failsafe_radio(bool b)
+{
+  // only act on changes
+    // -------------------
+    if(failsafe.radio != b) {
+
+        // store the value so we don't trip the gate twice
+        // -----------------------------------------------
+        failsafe.radio = b;
+
+        if (failsafe.radio == false) {
+            // We've regained radio contact
+            // ----------------------------
+            failsafe_radio_off_event();
+
+        }else{
+            // We've lost radio contact
+            // ------------------------
+            failsafe_radio_on_event();
+        }
+
+        // update AP_Notify
+        AP_Notify::flags.failsafe_radio = b;
+    }
+}
+
+// failsafe_radio_on_event - respond to radio contact being regained
```

Comment:
the comments are switched, right? failsafe_on_event() is called when the Failsafe turns on, not when it turns off

Reply:
yep...copy/paste error sunk in